client/signup/wpcom-login-form/index.jsx

```diff
@@ -4,6 +4,22 @@ import { Component } from 'react';
 
 const debug = debugFactory( 'calypso:signup:wpcom-login' );
 
+function getFormAction( redirectTo ) {
+	const subdomainRegExp = /^https?:\/\/([a-z0-9]*).wordpress.com/;
```

Comment:
The regexp looks wrong. Should it be more like this?
```js
/^https?:\/\/([a-z0-9]+)\.wordpress\.com(?:$|\/)/
```
1) require at least one character in the subdomain; 2) match dots instead of "any char"; 3) require either end of string or `/` after `.com`, to avoid matching domains like `wordpress.com.au`

Reply:
Agree with you both and have added @jsnajdr's suggestion in https://github.com/Automattic/wp-calypso/pull/68099/commits/1594753ccf7985b853e1504e85a0709dedc7a2b5. In general though this is something I'd like to manage at a more central place or shift responsibility to a package to avoid having multiple solutions (I'm sure something like this is used elsewhere in Calypso) which are lacking in one or more aspects like the previous one in this case.